pulser-core/pulser/sequence/sequence.py

```diff
@@ -728,6 +755,8 @@ def check_retarget(ch_obj: Channel) -> bool:
                         continue
                 if not strict:
                     channel_match[old_ch_name] = new_ch_id
+                    if ch_match_err.startswith(base_msg):
+                        ch_match_err = ""
```

Comment:
I'm not sure I understand why this is needed

Reply:
Well I had an issue that I only managed to solve this way: When switching from a device with 1 reusable dmm to a device with 1 non-reusable dmm, for a sequence configuring two dmms after having declared a global channel, the error message was "No match for channel dmm_0" whereas dmm_0 was matched, and the error message should be "No match for channel dmm_0_1" (which it is now, see test_switch_device_down)

Reply:
So I tried with moving the base message above, there was still the old error because the issue does not come from the base message but from the fact that an error message is defined while a corresponding channel is found. I can spend more time figuring out what is really happening, but I find it normal to have an error message because the DMM channels don't have the same type as the Rydberg/Raman/XY channels - however I don't fully understand why we did not witness this issue before, personally I have never tried to switch a device with a Rydberg and two Raman channels for instance.